EasyPost/_base/EasyPostClient.cs

```diff
@@ -21,57 +18,68 @@ public abstract class EasyPostClient
     {
         public readonly ClientConfiguration Configuration;
 
-        private readonly RestClient _restClient;
+        /// <summary>
+        ///     Gets the API key used by this client.
+        /// </summary>
+        public string ApiKeyInUse => Configuration.ApiKey; // public read-only property so users can audit the API key used by the client
+
+        /// <summary>
+        ///     Gets the base URL used by this client.
+        /// </summary>
+        public string ApiBaseInUse => Configuration.ApiBase; // public read-only property so users can audit the base URL used by the client
+
+        /// <summary>
+        ///     Gets the connect timeout in milliseconds used by this client.
+        /// </summary>
+        public int ConnectTimeoutMillisecondsInUse => (int)HttpClient.Timeout.TotalMilliseconds; // public read-only property so users can audit the connect timeout used by the client
+
+        /// <summary>
+        ///     Gets the custom HTTP client used by this client.
+        /// </summary>
+        public HttpClient? CustomHttpClientInUse => Configuration.HttpClient; // public read-only property so users can audit the custom HTTP client they set to be used by the client
+
+        /// <summary>
+        ///     Gets the prepared HTTP client used by this client for all API calls.
+        /// </summary>
```

Comment:
question: shouldn't a user only be able to get one or the other? Meaning, there is only ever a custom HTTP client or the default in use, never both?

Reply:
The `CustomHttpClient` is basically echoing what they passed in to the constructor, useful if they need to get that back (or use it to construct a new client, as we do for the Partner service).

The `PreparedClient` is basically the custom client or a standard client if one was not provided, with the timeout already set up. This is internal-only and therefore is purposefully inaccessible to the end user.